core/loader/json-feed.ts

```diff
@@ -0,0 +1,137 @@
+import type { TextResponse } from '../download.js'
+import type { OriginPost } from '../post.js'
+import { createPostsPage } from '../posts-page.js'
+import type { Loader } from './index.js'
+import { findAnchorHrefs, findLinksByType, toTime } from './utils.js'
+
+// https://www.jsonfeed.org/version/1.1/
+export type Author = {
+  avatar?: string
+  name?: string
+  url?: string
+}
+
+export type Item = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  banner_image?: string
+  content_html?: string
+  content_text?: string
+  date_modified?: string
+  date_published?: string
+  external_url?: string
+  id: string
+  image?: string
+  summary?: string
+  tags?: string[]
+  title?: string
+  url?: string
+}
+
+export type JsonFeed = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  description?: string
+  favicon?: string
+  feed_url?: string
+  home_page_url?: string
+  icon?: string
+  items: Item[]
+  next_url?: string
+  title: string
+  user_comment?: string
+  version: string
+}
+
+type ValidationRules = {
+  [key: string]: (value: unknown) => boolean
+}
+
+function isObjValid<ValidatedType>(
```

Comment:
What do you think if we rename it is `validate()` to highlight that main step here is rules based validation?

Reply:
Hm, I like it. I thought about adding a pointer in name to specify exactly what we validate, but the type and context around probably cover everything, and it's not necessary. I've updated it to just 'validate'.